spark/src/main/scala/geotrellis/spark/io/file/FileLayerMover.scala

```diff
@@ -0,0 +1,72 @@
+package geotrellis.spark.io.file
+
+import geotrellis.spark._
+import geotrellis.spark.io._
+import geotrellis.spark.io.avro._
+import geotrellis.spark.io.index._
+import geotrellis.spark.io.json._
+import geotrellis.raster.io.Filesystem
+import AttributeStore.Fields
+
+import spray.json.JsonFormat
+import org.apache.avro.Schema
+
+import scala.reflect.ClassTag
+import java.io.File
+
+object FileLayerMover {
+  def apply[K: JsonFormat: ClassTag, V: ClassTag, M: JsonFormat](sourceAttributeStore: FileAttributeStore, targetAttributeStore: FileAttributeStore): LayerMover[LayerId] =
+    new LayerMover[LayerId] {
+      def move(from: LayerId, to: LayerId): Unit = {
+        if(targetAttributeStore.layerExists(to))
+          throw new LayerExistsError(to)
+
+        val sourceMetadataFile = sourceAttributeStore.attributeFile(from, Fields.metaData)
+        if(!sourceMetadataFile.exists) throw new LayerNotFoundError(from)
+
+        // Read the metadata file out.
+        val (header, metadata, keyBounds, keyIndex, writerSchema) = try {
+          sourceAttributeStore.readLayerAttributes[FileLayerHeader, M, KeyBounds[K], KeyIndex[K], Schema](from)
+        } catch {
+          case e: AttributeNotFoundError => throw new LayerReadError(from).initCause(e)
+        }
+
+        // Delete the metadata file in the source
+        sourceMetadataFile.delete()
```

Comment:
This should probably happen only once we've written the metadata out successfully, it seems to me